core/src/test/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1238,7 +1238,7 @@ public void restartRandomDataNode(RestartCallback callback) throws Exception {
     /**
      * Restarts a random node in the cluster and calls the callback during restart.
      */
-    private void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
+    public void restartRandomNode(Predicate<NodeAndClient> predicate, RestartCallback callback) throws Exception {
```

Comment:
I don't think it's the intention to expose this to the outside..  I'm fine if we want to do this, but can we do it in another change? If you need to, please add a variant that uses both Predicate<Settings> filter and a callback